### DISCUSSION TITLE
ORM update execution time

### DISCUSSION THREAD

**Discussion by kigawas** [sqlalchemy/sqlalchemy#6813] [Usage Questions] [answered]
On latest sqlalchemy (1.4.20)

## Example
Say I have a User model (with an async session):

```python
user_a = await session.execute(select(User).where(User.username == 'aaa'))  # 1
user_a.password = 'abcdefg' # 2

await session.flush() # 3

user_b = await session.execute(select(User).where(User.username == 'bbb')) # 4
```
After some tests I found: If there is no `await session.flush()`, the `UPDATE` SQL will be executed at 4.

## Question
Is this set by default? Can I find any documentation about how/when the ORM changes get reflected by executing corresponding SQLs?

**Comment by zzzeek** [comment, marked as answer]
sure, understanding of the unit of work's process first and foremost is explained in the [tutorial](https://docs.sqlalchemy.org/en/14/tutorial/orm_data_manipulation.html#flushing) and then additional documentation on [flushing](https://docs.sqlalchemy.org/en/14/orm/session_basics.html#flushing)
